### DISCUSSION TITLE
How to Make a .Bin file with WifiConfig = 7

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11845] [Q&A] [answered]
Hello,
It is my understanding that, the command WifiConfig 7 will prevent WiFiManager from jumping into action and making the tasmota devices available using an unsecured access point in case of Home Wifi network failure, though this feature will remain active for Reset case.  

(WifiConfig 7 = set Wi-Fi Manager (web server at 192.168.4.1) as the current configuration tool restricted to reset settings only.) 

My question is, can I make this a default setting in Bin file through User_override_H file?  If not, and I enter this command in autoexecute bat file, then how do i flash this .bat file along with tasmota .Bin file? I don't want to have to login to each device and manually add the autoexec.bat file later.

Regards,

Navi

**Comment by meingraham** [comment, marked as answer]
Yes. Open the sample override file and you'll see the #define for that setting.

**Reply by NsinghP** [reply]
Thanks, I think I need to add the following custom line during compiling:

#define WIFI_CONFIG_TOOL       WIFI_MANAGER_RESET_ONLY 

Though the my_user_config_h file comment // states (default option: 4 - WIFI_RETRY) but that is not the case, it seems to me the default option is (0 - WIFI_RESTART).

Regards,

Navi

**Reply by NsinghP** [reply]
I compiled a .bin file with custom string "#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY" but when I give wificonfig command through console it still shows:

23:44:01.142 CMD: wificonfig
23:44:01.147 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}

Does this mean the default changes did not take effect? I was expecting it to be ":{"7" :"WIFI_MANAGER_RESET_ONLY"}}. What did i do wrong?

**Reply by barbudor** [reply]
The parameter you hardcoded will be used on a new device.
If you upgraded a device already configured  then it keeps the existing settings.
You can test using `reset 1` that will erase exist settings and enforced the new ones

**Reply by NsinghP** [reply]
Nope, still the same after reset 1,

00:32:41.340 CMD: wificonfig
00:32:41.345 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}

Here are my custom strings:

#define USE_UFILESYS
#define GUI_TRASH_FILE
#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[258,259,32,163,256,257,1,1,162,161,160,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define USE_EMULATION_HUE
#define EMULATION         EMUL_HUE
#define DEVICE_NAME       "First Room"  
#define FRIENDLY_NAME     "First Room Switch"
#define TIME_DST_OFFSET    +330          
#define TIME_STD_OFFSET    +330      
#define APP_TIMEZONE       99
#define USE_TASMOTA_DISCOVERY
#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY

Looks like last two strings don't matter. (TasmoManager still doesn't work)

**Comment by barbudor** [comment]
```
21:54:40.832 CMD: wificonfig
21:54:40.840 MQT: stat/nodemcu/WIFICONFIG = {"WifiConfig":{"7":"ManagerRst"}}
```
You probably have another #define somewhere that you forgot

**Reply by NsinghP** [reply]
> You probably have another #define somewhere that you forgot

Could it ibe #define USE_WIFI_CONFIG_TOOL ?

**Reply by NsinghP** [reply]
Ok so tried that and #define USE_WIFI_CONFIG_TOOL does not solve the problem.

**Reply by barbudor** [reply]
I mean another #define WIFI_CONFIG_TOOL that occurs later and override this one
Are you 100% sure the line is not commented?

**Reply by NsinghP** [reply]
Here is user_config_override.h file: 


[user_config_override.h-2.txt](https://github.com/arendst/Tasmota/files/6364076/user_config_override.h-2.txt)


I don't see it repeated anywhere.

**Reply by barbudor** [reply]
If you are using TasmoCompiler, please test with VSCode+PlatformIO or GitPod
or raise the question with TasmoCompiler
I had no problem setting `#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY` with VSCode

**Reply by NsinghP** [reply]
Yes, I am using TasmoCompiler. I have raised a question over there. https://github.com/benzino77/tasmocompiler/discussions/169

**Reply by NsinghP** [reply]
barbudor, A request to please see disscussion at https://github.com/benzino77/tasmocompiler/discussions/169 and help in figuring out what is that you are doing differently on VSCode+PlatformIO.--Thx Navi.

**Reply by benzino77** [reply]
@barbudor I think there is a problem with USER_TEMPLATE line. Can you please check this on your side? With this line included in `user_config_override.h` file my Wemos start behaving really strange.

**Reply by benzino77** [reply]
On serial console after reset I get this:
```
00:00:00.002 HDW: ESP8266EX
00:00:00.007 UFS: FlashFS mounted with 2000 kB free
00:00:00.054 CFG: Loaded from File, Count 7
00:00:00.059 FRC: Some settings have been r
```
and nothing else. It does not respond to any commands.

**Reply by barbudor** [reply]
Did you add the `\`for each `"` inside the string ?
It should be 
```
#define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[258,259,32,163,256,257,1,1,162,161,160,1,1,1],\"FLAG\":0,\"BASE\":18}"
```

Loading template manually in the GUI works fine

**Reply by NsinghP** [reply]
> 
> 
> Did you add the `\`for each `"` inside the string ?
> It should be
> 
> ```
> #define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[258,259,32,163,256,257,1,1,162,161,160,1,1,1],\"FLAG\":0,\"BASE\":18}"
> ```

Yes, that is not the problem, "\" go missing when i copy paste. (You can see that in .txt file i posted,it has the correct syntac with, all"\" in place)

**Reply by barbudor** [reply]
@NsinghP If you were adding the backticks that wouldn't happen :)
But my comment was to @benzino77

**Reply by NsinghP** [reply]
as a matter of fact they are not even showing up in the above message. "" = "inverted slash"

**Reply by barbudor** [reply]
Why does it works with me ?
![image](https://user-images.githubusercontent.com/2996042/115919533-3c06aa80-a479-11eb-9dd6-ee5770554c5b.png)

**Reply by NsinghP** [reply]
\\\\\\\I am not sure, it could be keyboard layout or something to do with my browser or OS wondow10 \\\\

**Reply by NsinghP** [reply]
"\"

**Reply by NsinghP** [reply]
it dosn's show up when i enclose it in "\".

**Reply by NsinghP** [reply]
I have done an erase/flash the problem still persists:

I am still gettings 22:15:09.468 CMD: wificonfig
22:15:09.473 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}

What is bugging me is that even if it is not accepting the custom string I am entering, why is it setting wificonfig to "0" when the default should be "4"

![Screenshot (119)](https://user-images.githubusercontent.com/69470381/115967671-865b5a80-a551-11eb-8355-818de59ed92b.png)

Command: esptool.py --port COM4 --baud 115200 --after no_reset write_flash --flash_mode dio 0x00000 C:\Users\NPHom\Desktop\ESP8266 FLASH\Tasmota\Custom BIN Files\18th run\firmware.bin --erase-all

esptool.py v2.6
Serial port COM4
Connecting....
Detecting chip type... ESP8266
Chip is ESP8266EX
Features: WiFi
MAC: 84:cc:a8:ae:e8:63
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 10.0s
Flash params set to 0x0240
Compressed 596368 bytes to 426240...
Wrote 596368 bytes (426240 compressed) at 0x00000000 in 37.9 seconds (effective 125.9 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode.

**Comment by barbudor** [comment]
Make be sure that you are not using an old "firmware.bin" instead of the latest downloaded
And download the new one with a different name such as firmware_wificonfig7.bin

I've done it again and everything works fine
```
00:00:00.001 HDW: ESP8266EX
00:00:00.130 UFS: FlashFS mounted with 2008 kB free
00:00:00.147 CFG: Use defaults
00:00:00.616 QPC: Reset
00:00:00.625 Project tasmota First Room Version 9.4.0.1(tasmota)-2_7_4_9(2021-04-24T17:56:53)
00:00:00.218 WIF: Connecting to AP1 BNETWLAN in mode 11n as tasmota_A6D683-5763...
00:00:07.004 WIF: Connected
00:00:07.256 HTP: Web server active on tasmota_A6D683-5763 with IP address 192.168.168.55
00:00:07.259 UPP: Multicast (re)joined
23:29:30.363 RSL: INFO1 = {"Info1":{"Module":"MYHOME","Version":"9.4.0.1(tasmota)","FallbackTopic":"cmnd/DVES_A6D683_fb/","GroupTopic":"cmnd/tasmotas/"}}
23:29:30.369 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota_A6D683-5763","IPAddress":"192.168.168.55"}}
23:29:30.379 RSL: INFO3 = {"Info3":{"RestartReason":"External System"}}
23:29:34.719 RSL: STATE = {"Time":"2021-04-24T23:29:34","Uptime":"0T00:00:13","UptimeSec":13,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"XXXXXX","BSSId":"00:14:D1:9E:8B:30","Channel":1,"RSSI":76,"Signal":-62,"LinkCount":1,"Downtime":"0T00:00:08"}}
23:29:43.039 CMD: wificonfig
23:29:43.044 RSL: RESULT = {"WifiConfig":{"7":"ManagerRst"}}
```

I had the same problem as benzimo77 so I replaced your template with all GPIO "1". I don't know why but your template prevent my Nodemcu from booting
==> this was because you are using the RX/TX so can't see the device booting on the serial

**Reply by barbudor** [reply]
My best bet is that you are not flashing the binary you think you should flash but an old one with WifiConfig 0 inside
Change the name of your template or DeviceName to be sure which firmware you get running

**Reply by NsinghP** [reply]
I am placing each new binary I compile in different folder with folder name indicating the times I have compiled before that i.e. folder name "18th run". In the latest compiles even the file system is not showing up, though the option was selected. I think I will have to figure out VScode/Platformio method and see if I have success.

#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define USE_EMULATION_HUE
#define EMULATION         EMUL_HUE
#define DEVICE_NAME       "First Room"  
#define FRIENDLY_NAME     "First Room Switch"
#define TIME_DST_OFFSET    +330          
#define TIME_STD_OFFSET    +330      
#define APP_TIMEZONE       99
#define USE_TASMOTA_DISCOVERY
#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY

#endif

**Reply by NsinghP** [reply]
I flash erased a brand new ESP8266 chip, same problem persisted. Then on this new chip, I reverted back to .bin I had compiled with 9.3.1.3, now my file system "Consoles" is back, but  wificonfig default value is still "0". (That doesn't make sense to me as default value should be "4", as I had not messed around with wifi_config_tool custom string when i compiled this binary.) It seems to me that wificonfig value "0" is permanently hardcoded in all my binaries and there is nothing I can do about it!). 

PS. I can change value to "7" by using console command wificonfig 7. That works, but If I enter the comand Reset 1, the value reverts back to "0" instead of "4".

Here are the strings I used for this binary:

#define USE_UFILESYS
#define GUI_TRASH_FILE
#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"MYHOME\",\"GPIO\":[258,259,32,163,256,257,1,1,162,161,160,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define USE_EMULATION_HUE
#define EMULATION    EMUL_HUE
#define DEVICE_NAME   "First Room"    
#define FRIENDLY_NAME   "First Room Switch"
#define TIME_DST_OFFSET    +330          
#define TIME_STD_OFFSET    +330      
#define APP_TIMEZONE       99

**Reply by sfromis** [reply]
I don't know if this is the issue, but I noticed above that you're flashing with the wrong options, specifically DIO instead of DOUT. Tasmotizer or ESP-Flasher much recommended for flashing, instead of tools like NodeMCU-PyFlasher. Try correct flashing.

**Reply by barbudor** [reply]
Did you checked in "Information" the build date of your binary ?
Did you tried to change the device name as I suggested ?
I am still convinced that either the upgrade did not worked or you are upgrading an old firmware
2 or 3 people tried with your settings and were successful

**Reply by NsinghP** [reply]
Hey Thanks. I Flash/Erased with two .bin files provided by Benzino, both of them, this time I also selected (DOUT) mode, but the problem persists. I have noticed another strange behavior that may be causing this problem. After Erase/Flash tasmota Wifi Access point asks for wifi credentials, when I enter my network name and passphrase, it is rejected the first time, then I have to reconnect to the the device access point a second time and it accepts the wifi credentials. I think between the first attempt and second attempt it is changing the default wificonfig to "0". There is no other explanation that I can think of that changes the value either from default "4" or from new "7" (if my custom string has been successfully incorporated into the binary) to "0".

Thanks for your help, I will kep trying until i figure this out.
![Screenshot (120)](https://user-images.githubusercontent.com/69470381/116000187-c895a200-a60c-11eb-8a1d-9f5bfbac9fa2.png)

Command: esptool.py --port COM4 --baud 115200 --after no_reset write_flash --flash_mode dout 0x00000 C:\Users\NPHom\Desktop\ESP8266 FLASH\Tasmota\Custom BIN Files\23rd run Benzino Desktop\firmware.bin --erase-all

esptool.py v2.6
Serial port COM4
Connecting....
Detecting chip type... ESP8266
Chip is ESP8266EX
Features: WiFi
MAC: 84:cc:a8:ae:e8:63
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 9.7s
Compressed 596368 bytes to 426239...
Wrote 596368 bytes (426239 compressed) at 0x00000000 in 37.9 seconds (effective 125.9 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode.

**Reply by NsinghP** [reply]
I have compiled again TasmoCompiler. (attaching compile messages in .txt file) Problem still persists and wificonfig gives status of "0" instead of "7". I am thinking that it may be Flash corruption issue. Nothing else makes sense. Also in the compile messages there is no mention of USE_TASMOTA_DISCOVERY, though my custom string includes #define Use_Tasmota_Discovery line. 
[Compile messages.txt](https://github.com/arendst/Tasmota/files/6409843/Compile.messages.txt)

**Reply by sfromis** [reply]
Looking a bit around in the Tasmota code, I found a potential source of setting WifConfig to 0. While the value `WIFI_RESTART` is not explicitly assigned anywhere, the code in `xdrv_01_webserver.ino` will save and restore the code and if this flow is broken, it could be restoring a value without it being saved.

An old value is saved in `HandleWifiConfiguration`:
[Web.old_wificonfig = TasmotaGlobal.wifi_state_flag;](https://github.com/arendst/Tasmota/blob/development/tasmota/xdrv_01_webserver.ino#L3334)

and restored after connection established in `FUNC_EVERY_SECOND`:
[Settings.sta_config = Web.old_wificonfig;](https://github.com/arendst/Tasmota/blob/development/tasmota/xdrv_01_webserver.ino#L3334)

Not thought through to flow, but I speculate that if a boot occurs between establishing wifi configuration and getting a connection, the value to be restored could be unassigned, which would often be 0.

**Reply by NsinghP** [reply]
with the newest compile though, the problem of "first attempt to connect with wifi failing is not happening". Also, when I activated MQTT server with broker.hivemq.com, TasmoManger app started working. So I think TasmoManger does not work on local network without entering a working MQTT server. (I can see bunch of devices on TasmoManager that are not mine, but since this is public MQTT server, I guess that is expected.)

![Screenshot (124)](https://user-images.githubusercontent.com/69470381/116789286-c67b8980-aacb-11eb-9a1d-4b473e39c97c.png)

**Reply by NsinghP** [reply]
Flashed a brand new chip. This time omitted custom string line "#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY" before compiling. Still wificonfig command is showing 0 instead of default 4. At this point, out of sheer frustration having spent so much time on the problem, I will be so much happier with default value of  4. I was trying to figure out how my device will behave with value set to 0. Does it mean that my device will keep on rebooting until it finds the current AP or alternate AP again? 

Tasmota website says:

0 = disable Wi-Fi Manager and reboot (used with alternate AP)

Lets say I want to associate it with new AP, will I have to factory reset the device?

Everyone's input is welcome and appreciated. 

Regards,

Navi

**Reply by barbudor** [reply]
> Does it mean that my device will keep on rebooting until it finds the current AP or alternate AP again?

yes


The fact that despite removing your `#define WIFI_CONFIG_TOOL` you still get 0 means that you absolutely underlooked the problem.
You should clear up everything, restart from a clean repository clone and add your changes one by one

**Reply by NsinghP** [reply]
yeah, thats what I'm thinking-- Start with downloading default tasmota.bin from the site and then customize one feature at a time. Lets see if i can figure out why "0" is replacing "4" for default wificonfig status. Thanks for your reply. ---Navi

**Reply by NsinghP** [reply]
allright, flashed latest tasmota.bin as is, without any customization to NodeMCU, using tasmotizer. Wificonfig command still reports status of "0". What am I missing?

Edit: Reporting Success!!! Reflashed with old tasmota.bin (8.5.0). Same NodeMCU was flashed using tasmotizer.  Wifconfig command now reports: 

18:40:18 CMD: wificonfig
18:40:18 RSL: stat/tasmota_CB20F0/RESULT = {"WifiConfig":{"4":"Retry"}}

I think the problem is with newer version of tasmota.bin

Edit 2: Reflashed same NodeMCU using tasmotiser and and latest tasmota.bin (9.4.0). The problem is back. Wificonfig command now reports:

18:48:37.970 CMD: wificonfig
18:48:37.974 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}

I am not sure why this is happening but I think problem is not with my hardware. Something is amiss with latest version of tasmota.

**Reply by barbudor** [reply]
Do you erase with each tasmotization?

**Reply by barbudor** [reply]
You must have made a voodoo sorcerer angry or something like that
Just tasmotized a NodeMCU with Tasmotizer 1.2, using Release 9.4.0 then sending only my wifi credential :
```
19:38:12.545 CMD: wificonfig
19:38:12.552 RSL: RESULT = {"WifiConfig":{"4":"Retry"}}
```

**Reply by barbudor** [reply]
Even without sending wifi credentials, running a serial terminal right after flashing I get:
```
00:00:00.002 HDW: ESP8266EX
00:00:00.034 CFG: Loaded from flash at F8, Count 4
00:00:00.039 QPC: Count 1
00:00:00.047 Project tasmota Tasmota Version 9.4.0(tasmota)-2_7_4_9(2021-04-23T10:07:22)
00:00:00.199 WIF: WifiManager active for 3 minutes
00:00:00.706 HTP: Web server active on tasmota_2AD017-4119 with IP address 192.168.4.1
00:00:06.916 QPC: Reset
wificonfig
00:00:08.761 CMD: wificonfig
00:00:08.765 RSL: RESULT = {"WifiConfig":{"4":"Retry"}}
```

**Reply by NsinghP** [reply]
Ha ha, I had the voodo doll ready, pins, needles and all, but then I stumbled across the solution in your reply! All i needed to do was send the wifi credentials through tasmotizer 1.2. I flashed a brand new NodeMCU, different from the one I used before. Instead of entering the wifi credentials through Tasmota Access point at 192.168.4.1, I  used "send config" feature of Tasmotizer1.2 to send the credentials to newly stock Tasmota.bin (9.4.0) flashed device. Now the console reports: 

09:23:52.422 CMD: wificonfig
09:23:52.427 RSL: RESULT = {"WifiConfig":{"4":"Retry"}}

That does not solve my problem entirely because I want to flash the device and let someone else (My niece in other state) configure the wifi credentials through Tasmota access point at 192.168.4.1.

I still think it is a software issue with latest Tasmota binary 9.4.0 that should not be difficult to fix.

 (Maybe I should learn coding/software programming, but these days I am busy with Voodo Magic, dolls, pins and needles. <--Just kidding.)

Edit: I was able to recreate the problem by reflashing same NodeMCU but this time using Tasmota Access Point 192.168.4.1 to enter wifi credentials. Wificonfig command reports: 

10:10:32.845 CMD: wificonfig
10:10:32.849 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}

Edit 2: I reflashed again and used Tasmotizer1.2 "Send Config" feature to load wifi credentials. Again the problem is solved. Wificonfig Command reports: 

10:26:42.861 CMD: wificonfig
10:26:42.866 RSL: RESULT = {"WifiConfig":{"4":"Retry"}}

Edit 3: Recreated the problem on different NodeMCU. Confirming that the problem is not with hardware.

**Reply by barbudor** [reply]
I can reproduce the same
```
00:00:00.001 HDW: ESP8266EX
00:00:00.017 CFG: Use defaults
00:00:00.445 QPC: Reset
00:00:00.455 Project tasmota Tasmota Version 9.4.0(tasmota)-2_7_4_9(2021-04-23T10:07:22)
00:00:00.752 WIF: WifiManager active for 3 minutes
00:00:00.261 HTP: Web server active on tasmota_2AD017-4119 with IP address 192.168.4.1
wificonfig
00:00:10.724 CMD: wificonfig
00:00:10.728 RSL: RESULT = {"WifiConfig":{"4":"Retry"}}
===> Entering wifi credential on the web page
00:00:58.864 WIF: Connecting to AP BNETWLAN as tasmota_2AD017-4119 ...
00:01:04.317 WIF: SSId1 BNETWLAN: Connected - IP Address 192.168.168.52
wificonfig
00:01:17.763 CMD: wificonfig
00:01:17.768 RSL: RESULT = {"WifiConfig":{"0":"Restart"}}
```

Please open a bug report on Github Issues

**Reply by NsinghP** [reply]
Could you be kind enough to do that please. I am not that good with software related stuff. It will be appreciated.

**Reply by barbudor** [reply]
Done : https://github.com/arendst/Tasmota/issues/12235

**Reply by NsinghP** [reply]
Thanks, I was looking at bug reporting page and find it rather intimidating.

**Reply by NsinghP** [reply]
Also noticed that if I flash the .bin file (9.4.0) I have compiled using my custom strings, then the option to "Send Config" does not work in tasmotizer1.2. then I am forced to use Tasmota AP to configure wifi credentials leading to"0" problem. I will trouble shoot this problem after "0" bug has been resolved.  I suspect this is due to Rx/Tx gpios being used. Here are my Strings:

#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"Hazel\",\"GPIO\":[256,259,32,163,257,258,1,1,161,160,162,1,1,1],\"FLAG\":0,\"BASE\":18}"
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE
#define DEVICE_NAME       "First Room"  
#define FRIENDLY_NAME     "First Room Switch"
#define TIME_DST_OFFSET    +330          
#define TIME_STD_OFFSET    +330      
#define APP_TIMEZONE       99
#define USE_TASMOTA_DISCOVERY
#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY

PS: Please ignore the missing \ (They are there, trust me, they just don't show up! I guess it is the Voodo mumbo jumbo :), I still haven't figured it out)

**Reply by ascillato2** [reply]
Fix Merged!

**Reply by NsinghP** [reply]
> 
> 
> Fix Merged!

Checked with latest development version 9.4.0.4 and the "0" problem is solved. Wificonfig command Reports: 

19:49:33.915 CMD: wificonfig
19:49:33.920 RSL: RESULT = {"WifiConfig":{"7":"ManagerRst"}}

Special Thanks to Barbudor and Ascillato2 for taking the time to understand this problem and then fixing it so fast. 

Regards, 

Navi

**Comment by NsinghP** [comment]
Yes, I am erasing with every flash. I am using tasmotizer 1.1. I  am
manually entering my WiFi credentials at the taamota access point
192.168.1.4. I think I will try with tasmotizer 1.2 tomorrow, but before
that let me get a voodo doll and some pins :)

**Comment by karoCB** [comment]
A current Tasmota release with wificonfig 7 (access point permanently on IP 192.168.4.1) can also be

simply via https://gitpod.io/#https://github.com/benzino77/tasmocompiler

#define WIFI_CONFIG_TOOL WIFI_MANAGER_RESET_ONLY

is set.

Separate individual client logins with different IP assignments from this AP server are then also possible.

But how the possible **8 simultaneous client sessions** can be achieved with an ESP32, I ask here as a question!

**Reply by sfromis** [reply]
For a **totally unrelated** question, do not use some old thread.

Anyway, Tasmota does not try to support a "full http server" with multi-threaded requests processed at the same time, but that may depend on your use case.
